### PR TITLE
Generating new schema

### PR DESCRIPTION
**Pull request by andrsGutirrz**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,10 +1,10 @@
 # json_yml_utility
 Read from JSON file, generate schema_test.yml from SQL models based on their metada-
 
-1) Make the script work: *
----- load/read models & manifest directories correctly
----- create schema test for all model, if no args passed on CLI
----- create schema test for one model, if by args passed 
+1) Make the script work: *
```
Comment (badilladrian) [inline README.md:4]: @andrsGutirrz  make the utility readme kindly

File: app/create_schema_test.py [hunk 2]
```diff
@@ -83,9 +83,9 @@ def _build_model_obj(self, model):
             "version": self.version,
             "models": [
                 OrderedDict({
-                    "name": model['name'],
-                    "description": model['description'],
-                    'original_file_path': model['original_file_path'],
+                    "name": model['metadata']['name'],  # TODO: confirm this is correct?
```
Comment (badilladrian) [inline app/create_schema_test.py:86]: Correct

File: app/create_schema_test.py [hunk 3]
```diff
@@ -83,9 +83,9 @@ def _build_model_obj(self, model):
...
+                    "description": '',  # model['description'],  # TODO: which is the field
```
Comment (badilladrian) [inline app/create_schema_test.py:87]: `        "source.hrdm_ingest.icims_raw.company": {
            "metadata": {
                "type": "BASE TABLE",
                "schema": "ICIMS",
                "name": "COMPANY", <<---- NAME
                "database": "HR_RAW_QA",
                "comment": "RAW TABLE FOR ICIMS COMPANY",  <<-- DESCRIPTION
                "owner": "_DB_HR_RAW_QA_ICIMS_OWN"
            },`

File: app/create_schema_test.py [hunk 4]
```diff
@@ -83,9 +83,9 @@ def _build_model_obj(self, model):
...
+                    'original_file_path': '',  # model['original_file_path'],  # TODO: which is the field
```
Comment (badilladrian) [inline app/create_schema_test.py:88]: The manifest provides where exactly the model is at the (whole) project structure (we have just a mini-fraction).

But, the same issue is presented on both scenarios-- Catalog does not contains the path of the .SQL model,
This is needed to know where to write the yml file

the method def write_yaml <- uses this path to match the yml with each model.

On the full extend project, as you can see the path structure can be more complex- so not all models are in the same directory, so the write_yaml NEEDS to know where to write the new generated yml accordingly to its model.

For this; I would suggest creating a new method that would go to app/ <- and make a dict with all the folder/files structures that i finds.

I.E.


app/
--- models (3)
--- other_models (5)
--- more_models (2)

Create a new method def generate_model_directory: --> output a dict with above structure

I suggest the following 
`for path, dirs, files in os.walk("."):
    print path
    for file in files:
        print os.path.join(path, file)
`

so there is no dependencies with the source-files to know where the models live in the project

File: app/create_schema_test.py [hunk 5]
```diff
@@ -169,28 +169,35 @@ def _build_columns_obj(self, model):
         return OrderedList(
             [
                 OrderedDict({
-                    'name': model['columns'][column]['name'],
-                    'description': model['columns'][column]['description'],
+                    'name': column['name'],
+                    'description': column.get(''),  # TODO: which is the field
```
Comment (badilladrian) [inline app/create_schema_test.py:173]: `            "columns": {
                "RAW_DATA": { 
                    "type": "VARIANT",   <---- this the DataType we need to add this! 
                    "index": 1,
                    "name": "RAW_DATA",  <---- this the colum name 
                    "comment": null  <---- this the description for column
                },
                "RAW_INSERTED_TIMESTAMP": {
                    "type": "TIMESTAMP_TZ",
                    "index": 2,
                    "name": "RAW_INSERTED_TIMESTAMP",
                    "comment": null
                },
                "RAW_FILENAME": {
                    "type": "TEXT",
                    "index": 3,
                    "name": "RAW_FILENAME",
                    "comment": null
                }
            },`
            
            Gotta iterate over each column

**Review by badilladrian — CHANGES_REQUESTED** [review]
Added some comments

File: app/create_schema_test.py [hunk 6]
```diff
@@ -169,28 +169,35 @@ def _build_columns_obj(self, model):
...
                         ]
                     ),
                     'meta': OrderedDict({
-                        'type': None,
-                        'privacy_classification': None,
-                        'ldm_model': None,
-                        'ldm_attribute': None,
-                        'datasource': None,
-                        'field': None
+                        'type': column['type'],
```
Comment (badilladrian) [inline app/create_schema_test.py:191]: correct

File: app/schema_test.yml [hunk 7]
```diff
@@ -0,0 +1,59 @@
+version: 2
```
Comment (badilladrian) [inline app/schema_test.yml:1]: @andrsGutirrz Here is the expected output for company

version: 2
models:
  - name: company
    description: 'RAW TABLE FOR ICIMS COMPANY'
    dbt_utils.recency:
      datepart: 
      field: 
      interval: 1
      tags: timeliness
      severity: warn
    dbt_expectations.expect_column_distinct_count_to_equal_other_table:
      column_name: 
      row_condition: 
      compare_model: 
      compare_column_name: 
      compare_row_condition: true
      tags: completeness
      severity: warn
    meta:
      slo:
        included: true
        delivery_time_ct:
        offset:
        is_static:
        contact:
        tag:
      ownership:
        team_id:
        ddo_delegate:
        ddo_delegate_ldap:
        subject_matter_expert:
        subject_matter_expert_ldap:
        approver:
        approver_ldap:
        approver_role:
      detailed_description:
        dataset_granularity:
        classification:
        classification_rationale:
        sensitive:
        is_degraded:
        comments:
      build_requirements:
        link_to_builder_code:
        link_to_orchestration_tool:
        link_to_design_document:
      origination:
        refresh_cadence:
        refresh_schedule:
      filters_and_limitations:
        standard_filters_at_consumption:
        general_limitations:
        field_discontinuities_irregularities:
        missing_data:
      data_access:
        access_controls:
        access_controls_review:
      data_quality:
        health_check:
  - name: raw_data
    description: 
    tests:
    - not_null:
        tags: validity
        severity: warn
    meta:
      type: variant
      privacy_classification: 
      ldm_model: 
      ldm_attribute: null
      datasource: 
      field: 
      comments: ''
  - name: raw_inserted_timestamp
    description: 
    tests:
    - not_null:
        tags: validity
        severity: warn
    meta:
      type: timestamp_tz
      privacy_classification: 
      ldm_model: 
      ldm_attribute: null
      datasource: 
      field: 
      comments: ''
  - name: raw_filename
    description: 
    tests:
    - not_null:
        tags: validity
        severity: warn
    meta:
      type: text
      privacy_classification: 
      ldm_model: 
      ldm_attribute: null
      datasource: 
      field: 
      comments: ''
      
      
     this is the goal output

Comment (andrsGutirrz) [reply app/schema_test.yml:1]: That's awesome, thanks

File: app/create_schema_test.py [hunk 8]
```diff
@@ -39,28 +39,28 @@ def start(self):
         try:
             logging.info("---- Schema Builder Utility Started! ----")
             self.models = self.get_models_from_manifest(self.model_selected)
-        except:
+        except FileNotFoundError as e:
             logging.error(f"Manifest.json file not found!")
         try:
             self.build_objs()
-            self.process_models()
+            # self.process_models()
```
Comment (badilladrian) [inline app/create_schema_test.py:46]: @andrsGutirrz  last not least--- uncomment this back!

def process_models as it has
                    f"Error processing {model_obj['models'][0]['name']} with error '{error}'")
                    
just to check if the access of the obj model is still the same, or this model_obj['models'][0]['name'] should be modified as well on this method for the new source catalog.json

File: README.md (hunk 1 above)
Comment (andrsGutirrz) [reply README.md]: Done

File: app/create_schema_test.py (hunk 2 above)
Comment (andrsGutirrz) [reply app/create_schema_test.py]: Done

File: app/create_schema_test.py (hunk 3 above)
Comment (andrsGutirrz) [reply app/create_schema_test.py]: Done

File: app/create_schema_test.py (hunk 4 above)
Comment (andrsGutirrz) [reply app/create_schema_test.py]: Using another approach. 
To create the route where we save the output yml, that path is the current path + model name

Comment (andrsGutirrz) [reply app/create_schema_test.py]: image

File: app/create_schema_test.py (hunk 6 above)
Comment (andrsGutirrz) [reply app/create_schema_test.py:191]: Done

File: app/create_schema_test.py (hunk 8 above)
Comment (andrsGutirrz) [reply app/create_schema_test.py]: No change required, this is running well

**Pull request closed without merging**